update-doi-status/src/main/java/no/unit/nva/doi/UpdateDoiStatusProcess.java

```diff
@@ -0,0 +1,100 @@
+        "Publication Identifier does not look like a id (URI)!";
+    public static final String FORWARD_SLASH = "/";
+    private static final Logger logger = LoggerFactory.getLogger(UpdateDoiStatusProcess.class);
+    public static final String UPDATED_PUBLICATION_FORMAT =
+        "Updated publication %s with doi: %s which was last modified: %s";
+    private final PublicationService publicationService;
+    private final DoiUpdateDto request;
+    private final Publication publication;
+
+    /**
+     * Start a UpdateDoiStatus process.
+     *
+     * @param publicationService publicationService for accessing publications
+     * @param request            {@code DoiUpdateHolder} representing a update doi status result.
+     */
+    public UpdateDoiStatusProcess(PublicationService publicationService, DoiUpdateHolder request) {
+        if (isInvalidPayloadFormat(request)) {
+            throw new IllegalArgumentException(
+                String.format(ERROR_BAD_DOI_UPDATE_HOLDER_FORMAT, request != null ? request.toJsonString() : null));
+        }
+        this.publicationService = publicationService;
+        this.request = request.getItem();
+
+        try {
+            UUID requestedPublicationId = extractPublicationFromRequest();
+            this.publication = publicationService.getPublication(requestedPublicationId);
+        } catch (ApiGatewayException e) {
+            throw DependencyRemoteNvaApiException.wrap(e);
+        }
+        validateInput();
+    }
+
+    /**
+     * Update DOI Status.
+     */
+    public void updateDoiStatus() {
+        try {
+            publication.setModifiedDate(request.getModifiedDate());
+            publication.setDoi(request.getDoi().get());
+            publicationService.updatePublication(publication.getIdentifier(), publication);
+            logPublicationDoiUpdate();
+        } catch (ApiGatewayException e) {
+            throw DependencyRemoteNvaApiException.wrap(e);
+        }
+    }
+
+    private void logPublicationDoiUpdate() {
+        logger.info(String.format(UPDATED_PUBLICATION_FORMAT,
+            publication.getIdentifier(),
+            publication.getDoi(),
+            publication.getModifiedDate()));
+    }
+
+    private boolean isInvalidPayloadFormat(DoiUpdateHolder request) {
+        return request == null || !request.hasItem() || !request.getItem().hasAllRequiredValuesSet();
+    }
+
+    private void validateInput() {
+        if (request.getModifiedDate().isAfter(Instant.now())) {
+            throw new IllegalStateException(MODIFIED_DOI_IS_IN_THE_FUTURE);
+        }
+        if (publication.getDoi() != null
+            && request.getDoi().isPresent()
+            && !publication.getDoi().toString().equals(request.getDoi().get())) {
+            throw new IllegalStateException(DOI_DOES_NOT_MATCH_DOI_IN_PUBLICATION);
+        }
+    }
+
+    private UUID extractPublicationFromRequest() {
+        String s = request.getPublicationId().toString();
```

Comment:
Maybe we should pass the identifier as well?

Reply:
`publicationService` works with UUIDs. We are eating a `publication ID (uri)` but I need to extract it to be able to use the `publicationService` for updating the persistence layer.

Reply:
Yes, I know. Maybe best if you didn't have to and it was already there in the event?

Reply:
Since we want to work with `id`s (JSON-LD), its correct for the payload to contain the `id`. I'm unsure if its correct to store or not store `id` instead of `identifier` in the storage, it's current implementation at least uses `identifier` now. I do not know how to proceed.

Reply:
The idea I had was to use identifier in addition to id